src/widgets/table/table.rs

```diff
@@ -564,8 +566,39 @@ impl<'a> Table<'a> {
         reason = "The name for this feature is not final and may change in the future",
         issue = "https://github.com/ratatui-org/ratatui/issues/536"
     )]
-    pub const fn segment_size(mut self, segment_size: SegmentSize) -> Self {
-        self.segment_size = segment_size;
+    #[deprecated(note = "please use flex instead of segment_size")]
```

Comment:
Nit: 
```suggestion
    #[deprecated(note = "You should use flex instead.")]
```

We can also have a more detailed message here. See the messages of other deprecated types.